### ISSUE TITLE
Can you change linter used with the Kite LSP?

### ISSUE THREAD

**Issue by e-cal** [kiteco/vim-plugin#354]
Is there a way to change what linter the Kite LSP uses? I would prefer to use pylint as it gives me warnings for Pep8 style guidelines as well as errors. 

Also, can you change when the linter runs? I prefer lint on save (default behavior of pylint + coc-python) as opposed to linting while I type if that is something that is configurable.

**Comment by airblade** [comment]
> Is there a way to change what linter the Kite LSP uses?

@metalogical I don't know the answer to this.

> Also, can you change when the linter runs? I prefer lint on save (default behavior of pylint + coc-python) as opposed to linting while I type if that is something that is configurable.

It's not configurable but you might be able to achieve what you want in a slightly hacky way.  I assume you have turned off the plugin's own completions (`let g:kite_completions=0`) in favour of LSP completions?  If so, I think (but haven't tested) that replacing these autocommands with `BufWritePre` ones might work:

https://github.com/kiteco/vim-plugin/blob/90d0e6cdffb214047548fc880619b0f4da04bd25/autoload/kite.vim#L137-L138

The tricky part is that they are set up anew on every `BufEnter` so until we make this overridable the simplest thing would be for you to edit the source code like this:

```diff
-    autocmd CursorHold,CursorHoldI   <buffer> call kite#events#event('selection')
-    autocmd TextChanged,TextChangedI <buffer> call kite#events#event('edit')
+    autocmd BufWritePre              <buffer> call kite#events#event('selection')
+    autocmd BufWritePre              <buffer> call kite#events#event('edit')
```

If it works I can make this configurable.  Let me know!

**Comment by e-cal** [comment]
Yeah I have kite completions disabled. Changing those lines doesn't appear to have any effect though, linting still only happens when text is inserted and all diagnostics disappear on write.
